ecosystem/indexer/src/default_processor.rs

```diff
@@ -0,0 +1,142 @@
+
+        let conn = self.connection_pool.get().map_err(|e| {
+            TransactionProcessingError::ConnectionPoolError((
+                anyhow::Error::from(e),
+                version,
+                self.name(),
+            ))
+        })?;
+
+        let tx_result = conn.transaction::<(), diesel::result::Error, _>(||{
+        aptos_logger::debug!(
+            "[default_processor] inserting 'transaction' version {} with hash {}",
+            version,
+            &transaction_model.hash
+        );
+        execute_with_better_error(
+            &conn,
+            diesel::insert_into(schema::transactions::table)
+                .values(&transaction_model)
+                .on_conflict(schema::transactions::dsl::hash)
+                .do_update()
+                .set(&transaction_model),
+        )
+        .expect("Error inserting row into database");
+
+        if let Some(tx_details_model) = maybe_details_model {
+            match tx_details_model {
+                Either::Left(ut) => {
+                    aptos_logger::debug!(
+                        "[default_processor] inserting 'user_transaction' version {} with hash {}",
+                        version,
+                        &transaction_model.hash
+                    );
+                    execute_with_better_error(
+                        &conn,
+                        diesel::insert_into(schema::user_transactions::table)
+                            .values(&ut)
+                            .on_conflict(schema::user_transactions::dsl::hash)
+                            .do_update()
+                            .set(&ut),
+                    )
+                    .expect("Error inserting row into database");
+                }
+                Either::Right(bmt) => {
+                    aptos_logger::debug!(
+                        "[default_processor] inserting 'block_metadata_transaction' version {} with hash {}",
+                        version,
+                        &transaction_model.hash
+                    );
+                    execute_with_better_error(
+                        &conn,
+                        diesel::insert_into(schema::block_metadata_transactions::table)
+                            .values(&bmt)
+                            .on_conflict(schema::block_metadata_transactions::dsl::hash)
+                            .do_update()
+                            .set(&bmt),
+                    )
+                    .expect("Error inserting row into database");
+                }
+            };
+        };
+
+        if let Some(events) = maybe_events {
+            execute_with_better_error(
+                &conn,
+                diesel::insert_into(schema::events::table)
+                    .values(events)
+                    .on_conflict_do_nothing(),
+            )
+            .expect("Error inserting row into database");
+        };
+            Ok(())
+        });
+
```

Comment:
something weird with spacing is happening here

Reply:
I agree, but this is from `cargo fmt`

Reply:
Yeah, sometimes `cargo fmt` is not intelligent enough to handle things, e.g., anonymous functions. You may have to try indenting things yourself 😄